pkgs/applications/science/biology/psipred/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib, stdenv, fetchFromGitHub, blast, tcsh, dbPath ? "uniref90.fasta", }:
```

Comment:
What's `dbPath` and why do we need to have it as an input?

Reply:
`DB` is an external amino acid sequence database, used by psipred.
This database may already prepare on the shared machine, and the path depends on its environment.
Since `dbPath` is [hard-coded](https://github.com/psipred/psipred/blob/5f68ee34f4b23daae6d1fea0b7a450ba40871e8b/BLAST%2B/runpsipredplus#L14) in `runpsipredplus` which is an executable script, you need to specify `dbPath` in a patch.
For example, `/shared/database/uniref90`, `/foo/bar/uniclust50` or  `${uniref90}/data/uniref90` if you make a derivation for uniref90.

Therefore, you should override the `dbPath` for your specific machine environment to use this package.

Reply:
Since it is a tcsh script, maybe it can be patched with an environment variable, that makes it simpler to switch databases. Actually, upstream should do that.

But it should have as default value `uniref90.fasta`, and I am not sure about how to set a default value for the variable in tcsh if it is not defined, like in bash.

Reply:
The usual for environment variables is to be uppercase, so it would be better to set it to  DB_PATH, but even that sounds too generic. Looks like BLAST_DATA or BLAST_DATA_BANK are the better choice, given the comment.

Reply:
Since README says SEQUENCE DATA BANK, I used SEQUENCE_DATA_BANK.
https://github.com/psipred/psipred